# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/AudioConduit.h
# Commit: 01b9c509cecf
# Full Hash: 01b9c509cecf7b5c0435b5ca063c3b4b492480d7
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 3
# Description:
#   Bug 1654112 - Fix AudioConduit::SendAudioFrame to use the new AudioSendStream API. r=ng
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102280
# ==============================================================================

diff -r 93367fdd03ec -r 01b9c509cecf dom/media/webrtc/libwebrtcglue/AudioConduit.h
--- a/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Fri Jan 29 15:38:10 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/AudioConduit.h	Fri Jan 29 15:38:43 2021 +0100
@@ -97,28 +97,14 @@
   /**
    * Function to deliver externally captured audio sample for encoding and
    * transport
-   * @param audioData [in]: Pointer to array containing a frame of audio
-   * @param lengthSamples [in]: Length of audio frame in samples in multiple of
-   *                             10 milliseconds
-   *                             Ex: Frame length is 160, 320, 440 for 16, 32,
-   *                             44 kHz sampling rates respectively.
-   *                             audioData[] should be of lengthSamples in
-   *                             size say, for 16kz sampling rate,
-   *                             audioData[] should contain 160 samples of
-   *                             16-bits each for a 10m audio frame.
-   * @param samplingFreqHz [in]: Frequency/rate of the sampling in Hz
-   *                             ( 16000, 32000 ...)
-   * @param capture_delay [in]:  Approx Delay from recording until it is
-   *                             delivered to VoiceEngine in milliseconds.
+   * @param frame [in]: AudioFrame in upstream's format for forwarding to the
+   *                    send stream. Ownership is passed along.
    * NOTE: ConfigureSendMediaCodec() SHOULD be called before this function can
    * be invoked. This ensures the inserted audio-samples can be transmitted by
-   * the conduit
+   * the conduit.
    */
-  MediaConduitErrorCode SendAudioFrame(const int16_t speechData[],
-                                       int32_t lengthSamples,
-                                       int32_t samplingFreqHz,
-                                       uint32_t channels,
-                                       int32_t capture_time) override;
+  MediaConduitErrorCode SendAudioFrame(
+      std::unique_ptr<webrtc::AudioFrame> frame) override;
 
   /**
    * Function to grab a decoded audio-sample from the media engine for